pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -624,6 +624,13 @@ public RebalanceResult rebalance(
           + "more servers.") @DefaultValue("false") @QueryParam("lowDiskMode") boolean lowDiskMode,
       @ApiParam(value = "Whether to use best-efforts to rebalance (not fail the rebalance when the no-downtime "
           + "contract cannot be achieved)") @DefaultValue("false") @QueryParam("bestEfforts") boolean bestEfforts,
+      @ApiParam(value = "How many maximum segment adds per server to update in the IdealState in each step. For "
+          + "non-strict replica group based assignment, this number will be the closest possible without splitting up "
+          + "a single segment's step's replicas across steps (so some servers may get fewer segments). For strict "
+          + "replica group based assignment, this is a per-server best effort value since each partition of a replica "
+          + "group must be moved as a whole and at least one partition in a replica group should be moved. A value of "
+          + "Integer.MAX_VALUE is used to indicate an unlimited batch size, which is the non-batching behavior.")
+      @DefaultValue("2147483647") @QueryParam("batchSizePerServer") int batchSizePerServer,
```

Comment:
Let's make default `0` or `-1` and treat non-positive value as unlimited

Reply:
done